core/unix/signal.c

```diff
@@ -2321,19 +2321,6 @@ handle_sigprocmask(dcontext_t *dcontext, int how, kernel_sigset_t *app_set,
             *error_code = EINVAL;
         return false;
     }
-    /* On MacOS, sigprocmask does not fail if the old sigset is not readable
-     * or not writeable.
-     */
-    if (IF_MACOS_ELSE(false, oset != NULL)) {
-        kernel_sigset_t safe_old_set;
-        /* Old sigset should be writable too. */
```

Comment:
> for the -no_intercept_all_signals case we do the wrong thing for a bad old-sigset but that we're going to remove support for that config.

Even for `-no_intercept_all_signals`, the current implementation does the right thing; all tests pass. 

For a bad old-sigset and `-no_intercept_all_signals`, the pre syscall checks will all pass, the syscall will be invoked and will fail on Linux (will succeed on Mac, because no bad-old-sigset failures happen on Mac), then the post-syscall handling will be invoked which will write the old-sigset in a TRY and will fault and return EFAULT again (no error on Mac, EFAULT on Linux).

Reply:
Note that in the above case, we need the syscall to be invoked. So that the kernel can update the app signal mask (sigprocmask updates the signal masks even when old-sigset is bad). Added a comment for this.

Reply:
Ack